### ISSUE TITLE
Foreach-Object -Parallel and "An item with the same key has already been added"

### ISSUE THREAD

**Issue by dwarfered** [microsoftgraph/msgraph-sdk-powershell#1597]
Hello,

It appears there's an intermittent issue when cmdlets are executed in parallel (say adding multiple users to a group concurrently).

Using the syntax Foreach-Object -Parallel { <cmdlet> }, there is intermittent appearance of:

"An item with the same key has already been added. Key: System.Net.Http.Headers.HeaderDescriptor".

Is there a switch or means of preventing the header (I assume bearer token) of being re-added and causing the thread safe violation?

**Comment by peombwa** [comment]
Thanks for following up on this.

Can please share a script that we can use to reproduce the error? Also, can please share with us the output of `$Error[0].Exception` immediately after the exception occurs.

**Comment by peombwa** [comment]
Consolidating the bug tracking in issue #1601.

We've narrowed down the issue to how `DefaultRequestHeaders` is being set. We will issue a release once the bug fix has been validated and tested.

Closing as duplicate of #1601.
